constraints/src/main/java/gov/nasa/jpl/aerie/constraints/time/IntervalAlgebra.java

```diff
@@ -224,6 +294,11 @@ static boolean endsStrictlyBefore(Interval x, Interval y) {
    * @return whether x ends when y begins, with no overlap and no gap
    */
   static boolean meets(Interval x, Interval y) {
+    // First try for a fast shortcut that doesn't require allocating a new interval.
+    if (!x.isEmpty() && !y.isEmpty()) {
+      return x.end.isEqualTo(y.start) && (x.endInclusivity != y.startInclusivity);
+    }
+
```

Comment:
I think rather than being a "shortcut", all of these are just faster ways of representing the same logic. Assuming these changes are all sound, anything that is not caught by the "shortcut" should return `false`, because one of the intervals is empty.

Reply:
Okay, for some reason I thought the existing implementation was being smart about empty and negative intervals.

Reply:
Yeah, not really. I think we were just focusing more on readability and delegation than performance.

Also there are no negative intervals; negative intervals are considered empty.